datarobot_provider/hooks/connections.py

```diff
@@ -0,0 +1,180 @@
+    :param datarobot_jdbc_conn_id: Connection ID, defaults to `datarobot_jdbc_default`
+    :type datarobot_jdbc_conn_id: str, optional
+    """
+
+    conn_name_attr = 'datarobot_jdbc_conn_id'
+    default_datarobot_jdbc_conn_name = 'datarobot_jdbc_default'
+    conn_type = 'datarobot_jdbc_datasource'
+    hook_name = 'DataRobot JDBC DataSource'
+
+    @staticmethod
+    def get_connection_form_widgets() -> Dict[str, Any]:
+        """Returns connection widgets to add to connection form."""
+        from flask_appbuilder.fieldwidgets import BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import StringField
+
+        return {
+            "datarobot_connection": StringField(
+                lazy_gettext('DataRobot Connection'),
+                widget=BS3TextFieldWidget(),
+                default='datarobot_default',
+            ),
+            "jdbc_driver": StringField(
+                lazy_gettext('JDBC Driver'),
+                widget=BS3TextFieldWidget(),
+                default='',
+            ),
+            "jdbc_url": StringField(
+                lazy_gettext('JDBC URL'),
+                widget=BS3TextFieldWidget(),
+                default='jdbc:',
+            ),
+        }
+
+    @staticmethod
+    def get_ui_field_behaviour() -> Dict:
+        """Returns custom field behaviour."""
+        return {
+            "hidden_fields": ['host', 'schema', 'port', 'extra'],
+            "relabeling": {},
+            "placeholders": {
+                'datarobot_connection': 'datarobot_default',
+                'jdbc_driver': '',
+                'jdbc_url': 'jdbc:',
+                'login': '',
+                'password': '',
+            },
+        }
+
+    def __init__(
+        self,
+        datarobot_jdbc_conn_id: str = default_datarobot_jdbc_conn_name,
+    ) -> None:
+        super().__init__()
+        self.datarobot_jdbc_conn_id = datarobot_jdbc_conn_id
+
+    def get_conn(self) -> (dict, DataStore):
+        """Initializes a DataRobot DataStore instance with associated credentials."""
+
+        conn = self.get_connection(self.datarobot_jdbc_conn_id)
+
+        datarobot_connection_id = conn.extra_dejson.get('datarobot_connection', '')
+
+        if not datarobot_connection_id:
+            raise AirflowException("datarobot_connection is not defined")
+
+        # Initialize DataRobot client by DataRobotHook
+        DataRobotHook(datarobot_conn_id=datarobot_connection_id).run()
+
+        jdbc_driver_name = conn.extra_dejson.get('jdbc_driver', '')
+
+        if not jdbc_driver_name:
+            raise AirflowException("jdbc_driver is not defined")
+
+        jdbc_url = conn.extra_dejson.get('jdbc_url', '')
+
+        if not jdbc_url:
+            raise AirflowException("jdbc_url is not defined")
+
+        if not conn.login:
+            raise AirflowException("login is not defined")
+
+        if not conn.password:
+            raise AirflowException("password is not defined")
```

Comment:
Can we have a configuration as dictionary with Python's standard data types? Then some [JSON Schema validator](https://pypi.org/project/jsonschema/) can be used to specify required fields and any additional rules.